apis/addons/v1alpha1/addons_types.go

```diff
@@ -313,6 +316,25 @@ type Addon struct {
 	Status AddonStatus `json:"status,omitempty"`
 }
 
+func (a *Addon) IsAvailable() bool {
+	return meta.IsStatusConditionTrue(a.Status.Conditions, Available)
+}
+
+func (a *Addon) SetUpgradePolicyStatus(val AddonUpgradePolicyValue) {
+	a.Status.UpgradePolicy = &AddonUpgradePolicyStatus{
+		ID:                 a.Spec.UpgradePolicy.ID,
```

Comment:
I don't the need for the ID anymore, but keeping it is harmless anyway... and adds up to the information for when we debug stuff

Reply:
never mind, let's keep it as it might change

Reply:
Yeah and I didn't want to create a breaking scenario by accident.